### PR TITLE
AccessToken-May-be-String

### PR DESCRIPTION
**Pull request by JRAndreassen**
There are some cases where the "AccessToken.expires_in" is returned as String.

This change makes the structure / Deserialization insensitive.
It will take either

### CODE REVIEW

**Comment by JRAndreassen** [comment]
Hmmm... It passed all the tests locally...
```text
> Executing task: cargo test --package graph-rs --test access_token_request --all-features -- client_credentials_test --exact --nocapture <
warning: value assigned to `next_link` is never read
  --> src\types\delta.rs:86:25
   |
86 |                         next_link = None;
   |                         ^^^^^^^^^
   |
   = note: `#[warn(unused_assignments)]` on by default
   = help: maybe it is overwritten before being read?

warning: 1 warning emitted

    Finished test [unoptimized + debuginfo] target(s) in 0.67s
     Running target\debug\deps\access_token_request-a4e9751ba7e2abdc.exe

running 1 test
test client_credentials_test ... ok

test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out
```

**Comment by sreeise** [comment]
I have to run the tests locally. Really annoying you cant get secrets without there being issues when you are not part of the code maintainer list.

**Comment by JRAndreassen** [comment]
Always something....
Let me know how it goes...

File: graph-oauth/src/accesstoken.rs [hunk 1]
```diff
@@ -514,3 +516,53 @@ impl fmt::Debug for AccessToken {
             .finish()
     }
 }
+
+#[allow(unused_imports, dead_code)]
```
Comment (sreeise) [inline graph-oauth/src/accesstoken.rs:520]: This isn't needed.

File: graph-oauth/src/accesstoken.rs [hunk 2]
```diff
@@ -514,3 +516,53 @@ impl fmt::Debug for AccessToken {
...
+
+
+    #[test]
+    pub fn test_deserialize() {
+        let _token: AccessToken = serde_json::from_str(ACCESS_TOKEN_INT).unwrap();
+        let _token: AccessToken = serde_json::from_str(ACCESS_TOKEN_STRING).unwrap();
+    }
+
+    #[test]
+    pub fn test_serialize_token() {
```
Comment (sreeise) [inline graph-oauth/src/accesstoken.rs:556]: We defineteley want to make sure both scenarios are covered but there are plenty of tests that already cover the expires_in field with it being an int. Can you remove this test and then your other test should be moved to https://github.com/sreeise/graph-rs/blob/master/tests/access_token_tests.rs

**Review by sreeise — CHANGES_REQUESTED** [review]
Tests look good locally. I commented on a few things that need too be cleaned up. Also, can you make sure to run `cargo fmt` for formatting.

File: graph-oauth/src/accesstoken.rs (hunk 1 above)
Comment (JRAndreassen) [reply graph-oauth/src/accesstoken.rs]: Done

File: graph-oauth/src/accesstoken.rs (hunk 2 above)
Comment (JRAndreassen) [reply graph-oauth/src/accesstoken.rs]: Done

**Review by sreeise — APPROVED** [review]

**Comment by sreeise** [comment]
Thanks for helping out @JRAndreassen!

**Comment by JRAndreassen** [comment]
@sreeise ,
Happy to help... 😎 

Now... Any plans on wrapping Teams ?
I only use it to time the connection for monitoring purposes ...

Thanks
   JR

**Comment by sreeise** [comment]
> @sreeise ,
> Happy to help... sunglasses
> 
> Now... Any plans on wrapping Teams ?
> I only use it to time the connection for monitoring purposes ...
> 
> Thanks
> JR

Yeah, actually thats part of what I am doing for codegen. The Teams API will be included in this branch https://github.com/sreeise/graph-rs/tree/api_codegen 

It is not here yet but it will be. That work is basically to be able to generate all of the Graph API's from the Open API specs that Microsoft provides. Its a good bit of reworking though so its taking some time. Should be done within two to three weeks, and a month at the most. Hopefully I can get it done sooner.

**Pull request merged**